// africastalking-ts/dist/products/index.old.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AfricasTalking = void 0;
var misc_1 = require("../utils/misc");
var utils_1 = require("../utils");
var airtime_1 = require("./airtime");
var application_1 = require("./application");
var payments_1 = require("./payments");
var sms_1 = require("./sms");
var token_1 = require("./token");
var voice_1 = require("./voice");
var AIRTIME = function (credentials) { return ({
    get sendAirtimeRequest() {
        misc_1.showDeprecationWarning('AfricasTalking().AIRTIME.sendAirtimeRequest()', 'new Client().sendAirtimeRequest()', 'minor');
        return airtime_1.sendAirtimeRequest(credentials);
    },
    get send() {
        misc_1.showDeprecationWarning('AfricasTalking().AIRTIME.send()', 'new Client().sendAirtimeRequest()', 'minor');
        return airtime_1.sendAirtimeRequest(credentials);
    },
}); };
var APPLICATION = function (credentials) { return ({
    get fetchApplicationData() {
        misc_1.showDeprecationWarning('AfricasTalking().APPLICATION.fetchApplicationData()', 'new Client().fetchApplicationData()', 'minor');
        return application_1.fetchApplicationData(credentials);
    },
    get fetchAccount() {
        misc_1.showDeprecationWarning('AfricasTalking().APPLICATION.fetchAccount()', 'new Client().fetchApplicationData()', 'minor');
        return application_1.fetchApplicationData(credentials);
    },
}); };
var PAYMENTS = function (credentials) { return (__assign({ get bankCheckoutCharge() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.bankCheckoutCharge()', 'new Client().bankCheckoutCharge()', 'minor');
        return payments_1.bankCheckoutCharge(credentials);
    },
    get bankCheckoutValidate() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.bankCheckoutValidate()', 'new Client().bankCheckoutValidate()', 'minor');
        return payments_1.bankCheckoutValidate(credentials);
    },
    get bankTransfer() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.bankTransfer()', 'new Client().bankTransfer()', 'minor');
        return payments_1.bankTransfer(credentials);
    },
    get cardCheckoutCharge() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.cardCheckoutCharge()', 'new Client().cardCheckoutCharge()', 'minor');
        return payments_1.cardCheckoutCharge(credentials);
    },
    get cardCheckoutValidate() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.cardCheckoutValidate()', 'new Client().cardCheckoutValidate()', 'minor');
        return payments_1.cardCheckoutValidate(credentials);
    },
    get mobileB2B() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.mobileB2B()', 'new Client().mobileB2B()', 'minor');
        return payments_1.mobileB2B(credentials);
    },
    get mobileB2C() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.mobileB2C()', 'new Client().mobileB2C()', 'minor');
        return payments_1.mobileB2C(credentials);
    },
    get mobileCheckout() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.mobileCheckout()', 'new Client().mobileCheckout()', 'minor');
        return payments_1.mobileCheckout(credentials);
    },
    get mobileData() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.mobileData()', 'new Client().mobileData()', 'minor');
        return payments_1.mobileData(credentials);
    },
    get fetchProductTransactions() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.fetchProductTransactions()', 'new Client().fetchProductTransactions()', 'minor');
        return payments_1.fetchProductTransactions(credentials);
    },
    get fetchWalletBalance() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.fetchWalletBalance()', 'new Client().fetchWalletBalance()', 'minor');
        return payments_1.fetchWalletBalance(credentials);
    },
    get fetchWalletTransactions() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.fetchWalletTransactions()', 'new Client().fetchWalletTransactions()', 'minor');
        return payments_1.fetchWalletTransactions(credentials);
    },
    get findTransaction() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.findTransaction()', 'new Client().findTransaction()', 'minor');
        return payments_1.findTransaction(credentials);
    },
    get topupStash() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.topupStash()', 'new Client().topupStash()', 'minor');
        return payments_1.topupStash(credentials);
    },
    get walletTransfer() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.walletTransfer()', 'new Client().walletTransfer()', 'minor');
        return payments_1.walletTransfer(credentials);
    },
    get checkout() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.checkout()', 'new Client().mobileCheckout()', 'minor');
        return payments_1.mobileCheckout(credentials);
    },
    get checkOut() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.checkOut()', 'new Client().mobileCheckout()', 'minor');
        return payments_1.mobileCheckout(credentials);
    },
    get payConsumer() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.payConsumer()', 'new Client().mobileB2C()', 'minor');
        return payments_1.mobileB2C(credentials);
    },
    get payBusiness() {
        misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS.payBusiness()', 'new Client().mobileB2B()', 'minor');
        return payments_1.mobileB2B(credentials);
    } }, utils_1.CONSTANTS)); };
var SMS = function (credentials) { return ({
    get createSubscription() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.createSubscription()', 'new Client().createSubscription()', 'minor');
        return sms_1.createSubscription(credentials);
    },
    get deleteSubscription() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.deleteSubscription()', 'new Client().deleteSubscription()', 'minor');
        return sms_1.deleteSubscription(credentials);
    },
    get fetchSubscription() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.fetchSubscription()', 'new Client().fetchSubscription()', 'minor');
        return sms_1.fetchSubscription(credentials);
    },
    get fetchMessages() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.fetchMessages()', 'new Client().fetchMessages()', 'minor');
        return sms_1.fetchMessages(credentials);
    },
    get sendSms() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.sendSms()', 'new Client().sendSms()', 'minor');
        return sms_1.sendSms(credentials);
    },
    get sendPremiumSms() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.sendPremiumSms()', 'new Client().sendPremiumSms()', 'minor');
        return sms_1.sendPremiumSms(credentials);
    },
    get send() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.send()', 'new Client().sendSms()', 'minor');
        return sms_1.sendSms(credentials);
    },
    get sendPremium() {
        misc_1.showDeprecationWarning('AfricasTalking().SMS.sendPremium()', 'new Client().sendPremiumSms()', 'minor');
        return sms_1.sendPremiumSms(credentials);
    },
}); };
var TOKEN = function (credentials) { return ({
    get createCheckoutToken() {
        misc_1.showDeprecationWarning('AfricasTalking().TOKEN.createCheckoutToken()', 'new Client().createCheckoutToken()', 'minor');
        return token_1.createCheckoutToken(credentials);
    },
    get generateAuthToken() {
        misc_1.showDeprecationWarning('AfricasTalking().TOKEN.generateAuthToken()', 'new Client().generateAuthToken()', 'minor');
        return token_1.generateAuthToken(credentials);
    },
}); };
var VOICE = function (credentials) { return ({
    get ActionBuilder() {
        misc_1.showDeprecationWarning('new AfricasTalking().VOICE.ActionBuilder()', 'new ActionBuilder()', 'minor');
        return utils_1.ActionBuilder;
    },
    get makeCall() {
        misc_1.showDeprecationWarning('AfricasTalking().VOICE.makeCall()', 'new Client().makeCall()', 'minor');
        return voice_1.makeCall(credentials);
    },
    get getNumQueuedCalls() {
        misc_1.showDeprecationWarning('AfricasTalking().VOICE.getNumQueuedCalls()', 'new Client().getNumQueuedCalls()', 'minor');
        return voice_1.getNumQueuedCalls(credentials);
    },
    get uploadMediaFile() {
        misc_1.showDeprecationWarning('AfricasTalking().VOICE.uploadMediaFile()', 'new Client().uploadMediaFile()', 'minor');
        return voice_1.uploadMediaFile(credentials);
    },
    get call() {
        misc_1.showDeprecationWarning('AfricasTalking().VOICE.call()', 'new Client().makeCall()', 'minor');
        return voice_1.makeCall(credentials);
    },
}); };
exports.AfricasTalking = function (credentials) {
    misc_1.showDeprecationWarning('AfricasTalking()', 'new Client()', 'minor');
    return {
        get AIRTIME() {
            misc_1.showDeprecationWarning('AfricasTalking().AIRTIME', 'new Client()', 'minor');
            return AIRTIME(credentials);
        },
        get APPLICATION() {
            misc_1.showDeprecationWarning('AfricasTalking().APPLICATION', 'new Client()', 'minor');
            return APPLICATION(credentials);
        },
        get PAYMENTS() {
            misc_1.showDeprecationWarning('AfricasTalking().PAYMENTS', 'new Client()', 'minor');
            return PAYMENTS(credentials);
        },
        get SMS() {
            misc_1.showDeprecationWarning('AfricasTalking().SMS', 'new Client()', 'minor');
            return SMS(credentials);
        },
        get TOKEN() {
            misc_1.showDeprecationWarning('AfricasTalking().TOKEN', 'new Client()', 'minor');
            return TOKEN(credentials);
        },
        get USSD() {
            misc_1.showDeprecationWarning('AfricasTalking().USSD', undefined, 'minor');
            throw new Error('USSD has been removed');
        },
        get VOICE() {
            misc_1.showDeprecationWarning('AfricasTalking().VOICE', 'new Client()', 'minor');
            return VOICE(credentials);
        },
        get ACCOUNT() {
            misc_1.showDeprecationWarning('AfricasTalking().ACCOUNT', 'new Client()', 'minor');
            return APPLICATION(credentials);
        },
        get PAYMENT() {
            misc_1.showDeprecationWarning('AfricasTalking().PAYMENT', 'new Client()', 'minor');
            return PAYMENTS(credentials);
        },
    };
};
